ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -460,6 +473,28 @@ PrjFS_Result PrjFS_WritePlaceholderFile(
         goto CleanupAndFail;
     }
     
+    if ((fileMode & (S_IXUSR|S_IXGRP|S_IXOTH)) != 0)
+    {
+        struct stat fileAttributes;
+        if (0 != lstat(fullPath, &fileAttributes))
+        {
+           LogWarning("PrjFS_WritePlaceholderFile: lstat failed on %s errno=%d, strerror=%s", fullPath, errno, strerror(errno));
+           result = PrjFS_Result_EIOError;
+           goto CleanupAndFail;
+        }
+
+        FsidInode fsidInode;
+        fsidInode.inode = fileAttributes.st_ino;
+        fsidInode.fsid = virtualizationRoot_fsid;
+
+        if (HydrateFile(fullPath, relativePath, fsidInode, 1, "placeholder") != PrjFS_Result_Success)
```

Comment:
Do we want to pass through the specific error here or explicitly EIOError as you’ve done?

Reply:
Under the covers it looks like it is all EIOError anyways.  

The EIOError is important because it signals to the provider that the affected file should be put in modified paths.  This means it will show up in 'git status' (which we want since it's likely different).

So in summary yes I think we want to explicity set this as PrjFS_Result_EIOError

Reply:
OK, sounds good!